src/main/kotlin/org/rust/ide/inspections/fixes/AddMainFnFix.kt

```diff
@@ -23,8 +22,7 @@ class AddMainFnFix(file: PsiElement) : LocalQuickFixAndIntentionActionOnPsiEleme
 
     override fun invoke(project: Project, file: PsiFile, editor: Editor?, startElement: PsiElement, endElement: PsiElement) {
         val function = file.add(RsPsiFactory(project).createFunction("fn main() { }")) as RsFunction
-        if (editor == null) return
         val offset = function.block?.lbrace?.textOffset ?: return
-        PsiNavigationSupport.getInstance().createNavigatable(project, file.virtualFile, offset + 1).navigate(true)
+        editor?.caretModel?.moveToOffset(offset + 1)
```

Comment:
Hm, interesting. Why didn't we do the same at the beginning? 🤔

Reply:
If I understand correctly, `PsiNavigationSupport` is for cases when file is not opened in the editor (and it will open it), for our cases when file is already opened, we can use `editor.caretModel`